pkg/filesystem/local_directory_unix.go

```diff
@@ -460,3 +463,53 @@ func (d *localDirectory) Apply(arg interface{}) error {
 		return syscall.EXDEV
 	}
 }
+
+func (d *localDirectory) Mount(mountpoint path.Component, source string, fstype string, options int) error {
+	if options != 0 {
+		return status.Error(codes.InvalidArgument, "Options are not yet supported in `mountat`.")
+	}
+	mfd, err := mountat(d.fd, fstype, source, mountpoint.String())
+	if err != nil {
+		return util.StatusWrap(err, "Mountat")
+	}
+	defer unix.Close(mfd)
+
+	return nil
+}
+
+// Fchdir changes the current working directory to the directory file descriptor.
+// If there is an error, it will be of type *PathError.
+func Fchdir(dfd int) error {
+	if e := syscall.Fchdir(dfd); e != nil {
+		return &fs.PathError{Op: "fchdir", Path: fmt.Sprintf("%d", dfd), Err: e}
+	}
+	return nil
+}
+
+// Uses `unmount` on relative pathnames with `fchdir`.
+func fchdir_unmountat(lock *sync.Mutex, dfd int, mountname string) error {
+	lock.Lock()
+	defer lock.Unlock()
+
+	store, err := os.Getwd()
+	if err != nil {
+		return util.StatusWrap(err, "Could not store the working directory %#v.")
+	}
+	defer os.Chdir(store)
+
+	err = Fchdir(dfd)
+	if err != nil {
+		return util.StatusWrapf(err, "Could not change working directory to %#v.", dfd)
+	}
+
+	err = unix.Unmount(mountname, 0)
+	if err != nil {
+		return util.StatusWrap(err, "Unmount")
+	}
+
+	return nil
+}
+
+func (d *localDirectory) Unmount(lock *sync.Mutex, mountpoint path.Component) error {
```

Comment:
I'd say, just keep this function in `mount_linux.go` as well. It's perfectly fine to have `localDirectory`'s `Unmount()` and `Mount()` methods in another file, without having all of the indirection.

Reply:
Okay, split it into `mount_{linux,unsupported}`, with the build tags for Linux and all other operating systems respectively.